lib/main.js

```diff
@@ -1,156 +1,98 @@
-      }),
-      atom.config.observe('linter-tslint.ignoreTypings', (ignoreTypings) => {
-        this.ignoreTypings = ignoreTypings;
-      }),
-      atom.workspace.observeTextEditors((textEditor) => {
-        // Marks each TypeScript editor with a CSS class so that
-        // we can enable commands only for TypeScript editors.
-        const rootScopes = textEditor.getRootScopeDescriptor().getScopesArray();
-        if (rootScopes.some(scope => grammarScopes.includes(scope))) {
-          atom.views.getView(textEditor).classList.add(editorClass);
-          textEditor.onDidSave(async () => {
-            if (atom.config.get('linter-tslint.fixOnSave')) {
-              await workerHelper.requestJob('fix', textEditor);
-            }
-          });
-        }
-      }),
-      atom.commands.add(`atom-text-editor.${editorClass}`, {
-        // Command subscriptions
-        'linter-tslint:fix-file': async () => {
-          const textEditor = atom.workspace.getActiveTextEditor();
+  getServerName() {
+    return 'TSLint';
+  }
 
-          if (!textEditor || textEditor.isModified()) {
-            // Abort for invalid or unsaved text editors
-            atom.notifications.addError('Linter-TSLint: Please save before fixing');
-            return;
-          }
+  getConnectionType() {
+    return 'ipc';
+  }
 
-          // The fix replaces the file content and the cursor can jump automatically
-          // to the beginning of the file, so save current cursor position
-          const cursorPosition = textEditor.getCursorBufferPosition();
+  getInitializeParams(projectPath, process) {
+    this.projectPath = projectPath;
 
-          try {
-            const results = await workerHelper.requestJob('fix', textEditor);
+    return {
+      processId: process.pid,
+      capabilities: {
+        workspace: {
+          configuration: true,
+        },
+      },
+      rootPath: projectPath,
+    };
+  }
 
-            const notificationText = results && results.length === 0 ?
-              'Linter-TSLint: Fix complete.' :
-              'Linter-TSLint: Fix attempt complete, but linting errors remain.';
+  startServerProcess(projectPath) {
+    this.projectPath = projectPath;
 
-            atom.notifications.addSuccess(notificationText);
-          } catch (err) {
-            atom.notifications.addWarning(err.message);
-          } finally {
-            // Restore cursor to the position before fix job
-            textEditor.setCursorBufferPosition(cursorPosition);
-          }
-        },
-      }),
+    return super.spawnChildNode(
+      [require.resolve('@keplersj/tslint-server/out/server')],
+      { cwd: projectPath, stdio: [null, null, null, 'ipc'] } // eslint-disable-line comma-dangle
     );
-
-    const createWorkerCallback = window.requestIdleCallback(() => {
-      this.worker = new Task(require.resolve('./worker.js'));
-      idleCallbacks.delete(createWorkerCallback);
+  }
+
+  preInitialization(connection) {
```

Comment:
I'd recommend moving most of this code - if not all of it - to the postInitialization hook as messages shouldn't be going back and forth until the initialization is complete.

The postInitialization function receives a server object which has a connection object on it as a property.

Reply:
Did this in my most recent commit. Thanks for the feedback!